src/browser/provider/pool.js

```diff
@@ -0,0 +1,145 @@
+import npm from 'npm';
+import { assignIn, flatten } from 'lodash';
+import promisify from '../../utils/promisify';
+import { PathBrowserProvider, LocalBrowserProvider, RemoteBrowserProvider } from './built-in';
+import BrowserProviderHost from './host';
+import SaucelabsBrowserProvider from './saucelabs';
+import { GeneralError } from '../../errors/runtime';
+import MESSAGE from '../../errors/runtime/message';
+import BrowserConnection from '../connection';
+
+var npmLoad = promisify(npm.load);
+var npmList = null;
+
+
+const BROWSER_PROVIDER_RE = /^(?:([^:]+):)?(.*)$/;
+const PROVIDER_MODULE_RE  = /^testcafe-provider-/;
+
+const providersCache  = {
+    'local':  new BrowserProviderHost(LocalBrowserProvider),
+    'path':   new BrowserProviderHost(PathBrowserProvider),
+    'remote': new BrowserProviderHost(RemoteBrowserProvider),
```

Comment:
BTW, what if I publish `testcafe-browser-provider-local` to npm? At first I thought we will publish built-in plugins to npm as separate packages, like we did for built-in reporters. But for providers I guess we should try to pick npm version first and only after that fallback to built-ins